### PR TITLE
chore(sdk): improve npm documentation

### LINKED ISSUE #4111: Improve npm documentation
## Description
- [ ] Add a bit of context to all the readmes of all public TS SDK packages explaining that they are for IOTA Rebased only and that this technology is only in Testnet and Devnet for now, not in Mainnet
- [ ] Cleanup debris migration info in kiosks: Remove the migration part
- [ ] Improve a bit the kiosk readme by adding instalation script and then a section with documentation that just points to the docs site
- [ ] Improve readmes to require min node v20

### PR DESCRIPTION
**Pull request by cpl121**
# Description of change

Please write a summary of your changes and why you made them.

## Links to any relevant issues

Closes #4111 

## Type of change

Choose a type of change, and delete any options that are not relevant.

- Bug fix (a non-breaking change which fixes an issue)
- Enhancement (a non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- Documentation Fix

## How the change has been tested

Describe the tests that you ran to verify your changes.

Make sure to provide instructions for the maintainer as well as any relevant configurations.

## Change checklist

Tick the boxes that are relevant to your changes, and delete any items that are not.

- [ ] I have followed the contribution guidelines for this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that new and existing unit tests pass locally with my changes

### CODE REVIEW

File: sdk/bcs/README.md [hunk 1]
```diff
@@ -1,5 +1,9 @@
 # BCS - Binary Canonical Serialization
 
+BCS is part of the **IOTA Rebased SDK**, designed specifically for interacting with the IOTA Rebased protocol. 
+
+> **Note**: This technology is currently available **only on Testnet and Devnet**, and is **not yet supported on Mainnet**.
```
Comment (marc2332) [inline sdk/bcs/README.md]: **technology** sounds weird

```suggestion
> **Note**: This package is currently only supported in **Testnet and Devnet**, and is **not yet supported in Mainnet**.
```

same for the rest

File: sdk/kiosk/README.md [hunk 2]
```diff
@@ -1,8 +1,41 @@
...
+npm i @iota/kiosk @iota/iota-sdk
+```
+
+To use the Kiosk SDK, you must create a [KioskClient](https://docs.iota.org/references/ts-sdk/kiosk/kiosk-client/introduction) instance.
+
+## Setup
+
+You can follow the example to create a KioskClient. The client currently supports MAINNET and TESTNET. View next section for usage in other networks.
+
+```
```
Comment (marc2332) [inline sdk/kiosk/README.md]:
```suggestion
```typescript
```

File: sdk/kiosk/README.md [hunk 3]
```diff
@@ -1,8 +1,41 @@
...
+
+```sh npm2yarn
+npm i @iota/kiosk @iota/iota-sdk
+```
+
+To use the Kiosk SDK, you must create a [KioskClient](https://docs.iota.org/references/ts-sdk/kiosk/kiosk-client/introduction) instance.
+
+## Setup
+
+You can follow the example to create a KioskClient. The client currently supports MAINNET and TESTNET. View next section for usage in other networks.
```
Comment (marc2332) [inline sdk/kiosk/README.md]:
```suggestion
You can follow this example to create a KioskClient.
```

File: sdk/kiosk/README.md [hunk 4]
```diff
@@ -1,8 +1,41 @@
...
+import { getFullnodeUrl, IotaClient } from '@iota/iota-sdk/client';
+
+// We need a IOTA Client. You can re-use the IotaClient of your project
+// (it's not recommended to create a new one).
+const client = new IotaClient({ url: getFullnodeUrl('testnet') });
+
+// Now we can use it to create a kiosk Client.
+const kioskClient = new KioskClient({
+    client,
+    network: Network.TESTNET,
```
Comment (marc2332) [inline sdk/kiosk/README.md]:
```suggestion
    network: Network.Testnet
```

File: sdk/kiosk/README.md [hunk 5]
```diff
@@ -1,8 +1,41 @@
...
+```
+
+To use the Kiosk SDK, you must create a [KioskClient](https://docs.iota.org/references/ts-sdk/kiosk/kiosk-client/introduction) instance.
+
+## Setup
+
+You can follow the example to create a KioskClient. The client currently supports MAINNET and TESTNET. View next section for usage in other networks.
+
+```
+import { KioskClient, Network } from '@iota/kiosk';
```
Comment (marc2332) [inline sdk/kiosk/README.md]:
```suggestion
import { KioskClient } from '@iota/kiosk';
```

File: sdk/kiosk/README.md [hunk 6]
```diff
@@ -1,8 +1,41 @@
...
+const client = new IotaClient({ url: getFullnodeUrl('testnet') });
+
+// Now we can use it to create a kiosk Client.
+const kioskClient = new KioskClient({
+    client,
+    network: Network.TESTNET,
+});
+```
+
+You can read the kioskClient documentation to query kiosk data [here](https://docs.iota.org/references/ts-sdk/kiosk/kiosk-client/querying).
```
Comment (marc2332) [inline sdk/kiosk/README.md]:
```suggestion
You can read the KioskClient documentation to query kiosk data [here](https://docs.iota.org/references/ts-sdk/kiosk/kiosk-client/querying).
```

**Review by marc2332 — CHANGES_REQUESTED** [review]

File: bridge/evm/README.md
Comment (marc2332) [inline bridge/evm/README.md:1]: `on` -> `in` imo

**Review by marc2332 — CHANGES_REQUESTED** [review]

**Review by marc2332 — APPROVED** [review]

File: sdk/kiosk/README.md (hunk 4 above)
Comment (brancoder) [reply sdk/kiosk/README.md]: Or `Network.Testnet` because that is the enum

**Review by brancoder — APPROVED** [review]
LGTM

**Review by bingyanglin — APPROVED** [review]

File: sdk/bcs/README.md [hunk 7]
```diff
@@ -1,5 +1,9 @@
 # BCS - Binary Canonical Serialization
 
+BCS is part of the **IOTA Rebased SDK**, designed specifically for interacting with the IOTA Rebased protocol.
+
+> **Note**: This package is currently supported in **only in Testnet and Devnet**, and is **not yet supported in Mainnet**.
```
Comment (begonaalvarezd) [inline sdk/bcs/README.md]:
```suggestion
> **Note**: This package is currently supported **only in Testnet and Devnet**, it is **not yet supported in Mainnet**.
```

File: sdk/create-dapp/README.md [hunk 8]
```diff
@@ -1,5 +1,9 @@
 # @iota/create-dapp
 
+create-dapp SDK is part of the **IOTA Rebased SDK**, designed specifically for interacting with the IOTA Rebased protocol.
```
Comment (begonaalvarezd) [inline sdk/create-dapp/README.md]:
```suggestion
`@iota/create-dapp` is part of the **IOTA Rebased SDK**, designed specifically for interacting with the IOTA Rebased protocol.
```

File: sdk/create-dapp/README.md [hunk 9]
```diff
@@ -1,5 +1,9 @@
...
+
+> **Note**: This package is currently supported in **only in Testnet and Devnet**, and is **not yet supported in Mainnet**.
```
Comment (begonaalvarezd) [inline sdk/create-dapp/README.md]:
```suggestion
> **Note**: This package is currently supported **only in Testnet and Devnet**, it is **not yet supported in Mainnet**.
```

File: sdk/dapp-kit/README.md [hunk 10]
```diff
@@ -1,5 +1,9 @@
 # IOTA dApp Kit
 
+IOTA dApp Kit is part of the **IOTA Rebased SDK**, designed specifically for interacting with the IOTA Rebased protocol.
+
+> **Note**: This package is currently supported in **only in Testnet and Devnet**, and is **not yet supported in Mainnet**.
```
Comment (begonaalvarezd) [inline sdk/dapp-kit/README.md]:
```suggestion
> **Note**: This package is currently supported **only in Testnet and Devnet**, it is **not yet supported in Mainnet**.
```

File: sdk/graphql-transport/README.md [hunk 11]
```diff
@@ -1,5 +1,9 @@
 # `@iota/graphql-transport`
 
+graphql-transport is part of the **IOTA Rebased SDK**, designed specifically for interacting with the IOTA Rebased protocol.
+
+> **Note**: This package is currently supported in **only in Testnet and Devnet**, and is **not yet supported in Mainnet**.
```
Comment (begonaalvarezd) [inline sdk/graphql-transport/README.md]:
```suggestion
> **Note**: This package is currently supported **only in Testnet and Devnet**, it is **not yet supported in Mainnet**.
```

File: sdk/kiosk/README.md [hunk 12]
```diff
@@ -1,8 +1,41 @@
 # Kiosk SDK
 
+Kiosk SDK is part of the **IOTA Rebased SDK**, designed specifically for interacting with the IOTA Rebased protocol.
+
+> **Note**: This package is currently supported in **only in Testnet and Devnet**, and is **not yet supported in Mainnet**.
```
Comment (begonaalvarezd) [inline sdk/kiosk/README.md]:
```suggestion
> **Note**: This package is currently supported **only in Testnet and Devnet**, it is **not yet supported in Mainnet**.
```

File: sdk/ledgerjs-hw-app-iota/README.md [hunk 13]
```diff
@@ -4,6 +4,10 @@
 
 # ledgerjs-hw-app-iota
 
+ledgerjs-hw-app-iota is part of the **IOTA Rebased SDK**, designed specifically for interacting with the IOTA Rebased protocol.
+
+> **Note**: This package is currently supported in **only in Testnet and Devnet**, and is **not yet supported in Mainnet**.
```
Comment (begonaalvarezd) [inline sdk/ledgerjs-hw-app-iota/README.md]:
```suggestion
> **Note**: This package is currently supported **only in Testnet and Devnet**, it is **not yet supported in Mainnet**.
```

File: sdk/typescript/README.md [hunk 14]
```diff
@@ -5,6 +5,10 @@ For more complete docs, visit the
 
 # IOTA TypeScript SDK
 
+TypeScript SDK is part of the **IOTA Rebased SDK**, designed specifically for interacting with the IOTA Rebased protocol.
```
Comment (begonaalvarezd) [inline sdk/typescript/README.md]:
```suggestion
`@iota/iota-sdk` is part of the **IOTA Rebased SDK**, designed specifically for interacting with the IOTA Rebased protocol.
```

File: sdk/typescript/README.md [hunk 15]
```diff
@@ -5,6 +5,10 @@ For more complete docs, visit the
...
+
+> **Note**: This package is currently supported in **only in Testnet and Devnet**, and is **not yet supported in Mainnet**.
```
Comment (begonaalvarezd) [inline sdk/typescript/README.md]:
```suggestion
> **Note**: This package is currently supported **only in Testnet and Devnet**, it is **not yet supported in Mainnet**.
```

File: sdk/bcs/README.md [hunk 16]
```diff
@@ -1,5 +1,9 @@
 # BCS - Binary Canonical Serialization
 
+BCS is part of the **IOTA Rebased SDK**, designed specifically for interacting with the IOTA Rebased protocol.
```
Comment (begonaalvarezd) [inline sdk/bcs/README.md]:
```suggestion
`@iota/bcs` is part of the **IOTA Rebased SDK**, designed specifically for interacting with the IOTA Rebased protocol.
```

File: sdk/dapp-kit/README.md [hunk 17]
```diff
@@ -1,5 +1,9 @@
 # IOTA dApp Kit
 
+IOTA dApp Kit is part of the **IOTA Rebased SDK**, designed specifically for interacting with the IOTA Rebased protocol.
```
Comment (begonaalvarezd) [inline sdk/dapp-kit/README.md]:
```suggestion
`@iota/dapp-kit` is part of the **IOTA Rebased SDK**, designed specifically for interacting with the IOTA Rebased protocol.
```

File: sdk/graphql-transport/README.md [hunk 18]
```diff
@@ -1,5 +1,9 @@
 # `@iota/graphql-transport`
 
+graphql-transport is part of the **IOTA Rebased SDK**, designed specifically for interacting with the IOTA Rebased protocol.
```
Comment (begonaalvarezd) [inline sdk/graphql-transport/README.md]:
```suggestion
`@iota/graphql-transport` is part of the **IOTA Rebased SDK**, designed specifically for interacting with the IOTA Rebased protocol.
```

File: sdk/kiosk/README.md [hunk 19]
```diff
@@ -1,8 +1,41 @@
 # Kiosk SDK
 
+Kiosk SDK is part of the **IOTA Rebased SDK**, designed specifically for interacting with the IOTA Rebased protocol.
```
Comment (begonaalvarezd) [inline sdk/kiosk/README.md]:
```suggestion
`@iota/kiosk` is part of the **IOTA Rebased SDK**, designed specifically for interacting with the IOTA Rebased protocol.
```

File: sdk/ledgerjs-hw-app-iota/README.md [hunk 20]
```diff
@@ -4,6 +4,10 @@
 
 # ledgerjs-hw-app-iota
 
+ledgerjs-hw-app-iota is part of the **IOTA Rebased SDK**, designed specifically for interacting with the IOTA Rebased protocol.
```
Comment (begonaalvarezd) [inline sdk/ledgerjs-hw-app-iota/README.md]:
```suggestion
`@iota/ledgerjs-hw-app-iota` is part of the **IOTA Rebased SDK**, designed specifically for interacting with the IOTA Rebased protocol.
```

File: sdk/wallet-standard/README.md [hunk 21]
```diff
@@ -1,5 +1,9 @@
 # `@iota/wallet-standard`
 
+wallet-standard is part of the **IOTA Rebased SDK**, designed specifically for interacting with the IOTA Rebased protocol.
```
Comment (begonaalvarezd) [inline sdk/wallet-standard/README.md]:
```suggestion
`@iota/wallet-standard` is part of the **IOTA Rebased SDK**, designed specifically for interacting with the IOTA Rebased protocol.
```

File: sdk/wallet-standard/README.md [hunk 22]
```diff
@@ -1,5 +1,9 @@
...
+
+> **Note**: This package is currently supported in **only in Testnet and Devnet**, and is **not yet supported in Mainnet**.
```
Comment (begonaalvarezd) [inline sdk/wallet-standard/README.md]:
```suggestion
> **Note**: This package is currently supported **only in Testnet and Devnet**, it is **not yet supported in Mainnet**.
```

**Review by begonaalvarezd — CHANGES_REQUESTED** [review]

File: sdk/kiosk/README.md (hunk 4 above)
Comment (marc2332) [reply sdk/kiosk/README.md]: Yes, should be `Network.Testnet`

File: sdk/kiosk/README.md [hunk 23]
```diff
@@ -1,8 +1,41 @@
...
+
+You can follow this example to create a KioskClient.
+
+```typescript
+import { KioskClient } from '@iota/kiosk';
+import { getFullnodeUrl, IotaClient, Network } from '@iota/iota-sdk/client';
+
+// We need a IOTA Client. You can re-use the IotaClient of your project
+// (it's not recommended to create a new one).
+const client = new IotaClient({ url: getFullnodeUrl(Network.TESTNET) });
```
Comment (marc2332) [inline sdk/kiosk/README.md]:
```suggestion
const client = new IotaClient({ url: getFullnodeUrl(Network.Testnet) });
```

**Review by marc2332 — CHANGES_REQUESTED** [review]

**Comment by msarcev** [comment]
Also please check throughout our `package.json`s if we have the node engine constraint set to `< 20` then.
Like i see one like that in `wallet-dashboard`
```
    "engines": {
        "node": ">= v18.17.0"
    },

```

**Review by begonaalvarezd — APPROVED** [review]

**Review by marc2332 — APPROVED** [review]

**Pull request merged**